File: server/src/engine/factories/wallfactory.ts
import Transform = require("../components/transform");
import Render = require("../components/render");
import EntityManager = require("../entitymanager");

class WallFactory {
    static new(em: EntityManager, position: Vector2): number {
        var wall = em.createEntity("Wall");
        em.addComponent(wall, new Transform(position, 100));
        em.addComponent(wall, new Render("Dungeon wall"));
        return wall;
    }
}

export = WallFactory

File: sharpServer/Engine/Factories/WallFactory.cs
namespace SharpServer.Engine.Factories
{
    public static class WallFactory
    {
        public static uint Create(Vector2 position)
        {
            var wall = EntityBaseFactory.Create(position, "Wall");
            EntityManager.AddComponents(wall, 
                new Render("Dungeon wall"),
                new Shape(100));
            return wall;
        }
    }
}